src/rust/engine/process_execution/src/remote.rs

```diff
@@ -844,15 +846,80 @@ fn maybe_add_workunit(
   }
 }
 
-pub fn make_execute_request(
+fn make_wrapper_for_append_only_caches(
+  caches: &BTreeMap<CacheName, RelativePath>,
+  base_path: &str,
+) -> Result<String, String> {
+  let mut script = String::new();
```

Comment:
Probably over-optimization, but maybe pre-allocate based on the length of caches?

Reply:
Or just preallocate something like 512 bytes.